reana_workflow_engine_yadage/cli.py

```diff
@@ -68,25 +70,22 @@ def run_yadage_workflow(workflow_uuid,
     os.umask(REANA_WORKFLOW_UMASK)
 
     cap_backend = setupbackend_fromstring('fromenv')
-    toplevel = os.getcwd()
-    workflow = None
-
-    if workflow_json:
-        # When `yadage` is launched using an already validated workflow file.
-        workflow_kwargs = dict(workflow_json=workflow_json)
-    elif workflow:
-        # When `yadage` resolves the workflow file from a remote repository:
-        # i.e. github:reanahub/reana-demo-root6-roofit/workflow.yaml
-        workflow_kwargs = dict(workflow=workflow, toplevel=toplevel)
-    elif workflow_file:
-        workflow_file_abs_path = os.path.join(
-            workflow_workspace, workflow_file)
-        if os.path.exists(workflow_file_abs_path):
+    workflow_file_abs_path = os.path.join(workflow_workspace, workflow_file)
+    publisher = None
+    try:
+        if not os.path.exists(workflow_file_abs_path):
+            log.error(f'Workflow file {workflow_file_abs_path} does not exist')
+        else:
             schema_name = 'yadage/workflow-schema'
             schemadir = None
-
+            toplevel = operational_options.get('toplevel')
+            if toplevel:
+                if not toplevel.startswith('github:'):
+                    toplevel = os.path.join(workflow_workspace, toplevel)
```

Comment:
We could move this logic to a different function and change the callback for the `--operational-options` option. For example:

```console
@click.option('--operational-options',
              help='Options to be passed to the workflow engine'
                   ' (e.g. initdir).',
              callback=load_yadage_operational_options)
```

And `load_yadage_operational_options` would load these defaults if not set.

Reply:
Is it possible to access the `workflow_workspace` from the callback?

Reply:
Auto reply: Yes, through `ctx.params['workflow_workspace']`.